### PR TITLE
CPU Manager panics on state initialization error.

### PR DESCRIPTION
**Pull request by ConnorDoyle**
**What this PR does / why we need it**:

- CPU Manager panics on state initialization error.
- Update unit tests accordingly.
- Minor related cleanup in `state_file.go`.

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

/kind bug
/sig node
/priority important-soon
Blocks #52031
/assign @balajismaniam 
cc @flyingcougar

### CODE REVIEW

File: pkg/kubelet/cm/cpumanager/state/state_file_test.go [hunk 1]
```diff
@@ -77,13 +77,15 @@ func TestFileStateTryRestore(t *testing.T) {
...
 	}{
 		{
-			"Invalid JSON - empty file",
+			"Invalid JSON - one byte file",
 			"\n",
 			"none",
-			"state file: could not unmarshal, corrupted state file",
+			"[cpumanager] state file: unable to restore state from disk (unexpected end of JSON input)",
+			true,
 			&stateMemory{
```
Comment (flyingcougar) [inline pkg/kubelet/cm/cpumanager/state/state_file_test.go]: nit: `expectedState` not needed at all when `expPanic` is true

**Review by flyingcougar — APPROVED** [review]
This should simplify `reconcile()` loop. Thx @ConnorDoyle .

File: pkg/kubelet/cm/cpumanager/state/state_file_test.go (hunk 1 above)
Comment (ConnorDoyle) [reply pkg/kubelet/cm/cpumanager/state/state_file_test.go]: Good point, will update

Comment (ConnorDoyle) [reply pkg/kubelet/cm/cpumanager/state/state_file_test.go]: Updated

**Comment by balajismaniam** [comment]
The errors are unrelated to this PR.
/retest

**Comment by ConnorDoyle** [comment]
@derekwaynecarr could we get this one into the 1.9 milestone?

**Comment by flyingcougar** [comment]
flaked due to #56262

**Comment by flyingcougar** [comment]
flake fixed
/retest

**Pull request merged**